### ISSUE TITLE
Why DeepMIM work?

### ISSUE THREAD

**Issue by jsrdcht** [OliverRensu/DeepMIM#1]
Please give some high-level insight?

**Comment by raymin0223** [comment]
Hi, thanks for the author’s great work!

From my understanding, DeepMIM regularizes the intermediate features to contain information for reconstructing original images, like the last feature.
This leads to several benefits, including **1)** regularization effects, **2)** addressing vanishing gradients, or **3)** target-awared and aligned information in intermediate features.
_i.e._, the mutual information between intermediate and last features may be expected to increase, given that both have the same targets.

I guess the analysis in our paper ([Self-Contrastive Learning](https://arxiv.org/abs/2106.15499)) could be applied similarly to this work.
As our loss function makes intermediate layers ($T$) to output similar representations to the last layer ($F$) via supervised contrastive loss, we have proved some theoretical guarantees regarding the MI between them.

In the unsupervised case, we demonstrated that self-contrasting guarantees the lower bound for $I(T, F)$, and the training was effective while freezing $F$ to prevent it from following $T$.
Even though they set the targets as original images, rather than the last representations (_i.e._, reconstructed patches from the last layer), I hope that our work could provide some value for the authors. :)

**Comment by jsrdcht** [comment]
> I guess the analysis in our paper ([Self-Contrastive Learning](https://arxiv.org/abs/2106.15499)) could be applied similarly to this work. As our loss function makes intermediate layers (T) to output similar representations to the last layer (F) via supervised contrastive loss, we have proved some theoretical guarantees regarding the MI between them.
> 
> In the unsupervised case, we demonstrated that self-contrasting guarantees the lower bound for I(T,F), and the training was effective while freezing F to prevent it from following T. Even though they set the targets as original images, rather than the last representations (_i.e._, reconstructed patches from the last layer), I hope that our work could provide some value for the authors. :)

I tend to think that this technology is "correct nonsense". Since ViT maintains feature resolution, the mutual information between high and low layers can be well preserved. Why do we still need additional objectives to help? The reconstruction objective itself requires maximizing mutual information to complete the task.

**Comment by raymin0223** [comment]
In my opinion, large models are capable of learning more semantic information in general, as the accuracy tends to improve with the addition of more transformer blocks.
This suggests that the learned information from earlier or deeper layers differs from one another similar to previous findings on convolution networks.
Thus, I think that DeepMIM can exploit information from various depths of networks via auxiliary MIM losses on earlier layers.

**Comment by raymin0223** [comment]
> > I guess the analysis in our paper ([Self-Contrastive Learning](https://arxiv.org/abs/2106.15499)) could be applied similarly to this work. As our loss function makes intermediate layers (T) to output similar representations to the last layer (F) via supervised contrastive loss, we have proved some theoretical guarantees regarding the MI between them.
> > In the unsupervised case, we demonstrated that self-contrasting guarantees the lower bound for I(T,F), and the training was effective while freezing F to prevent it from following T. Even though they set the targets as original images, rather than the last representations (_i.e._, reconstructed patches from the last layer), I hope that our work could provide some value for the authors. :)
> 
> I tend to think that this technology is "correct nonsense". Since ViT maintains feature resolution, the mutual information between high and low layers can be well preserved. Why do we still need additional objectives to help? The reconstruction objective itself requires maximizing mutual information to complete the task.

In my opinion, large models are capable of learning more semantic information in general, as the accuracy tends to improve with the addition of more transformer blocks.
This suggests that the learned information from earlier or deeper layers differs from one another similar to previous findings on convolution networks.
Thus, I think that DeepMIM can exploit information from various depths of networks via auxiliary MIM losses on earlier layers.
